integration-tests/mongodb-panache/src/main/java/io/quarkus/it/mongodb/panache/axle/book/ReactiveBookEntityResource.java

```diff
@@ -0,0 +1,118 @@
+package io.quarkus.it.mongodb.panache.axle.book;
+
+import java.net.URI;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+
+import javax.annotation.PostConstruct;
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.bson.types.ObjectId;
+import org.jboss.logging.Logger;
+import org.jboss.resteasy.annotations.SseElementType;
+import org.reactivestreams.Publisher;
+
+import io.quarkus.panache.common.Parameters;
+import io.quarkus.panache.common.Sort;
+
+@Path("/axle/books/entity")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class ReactiveBookEntityResource {
+    private static final Logger LOGGER = Logger.getLogger(ReactiveBookEntityResource.class);
+
+    @PostConstruct
+    void init() {
+        String databaseName = ReactiveBookEntity.mongoDatabase().getName();
+        String collectionName = ReactiveBookEntity.mongoCollection().getNamespace().getCollectionName();
+        LOGGER.infov("Using BookEntity[database={0}, collection={1}]", databaseName, collectionName);
+    }
+
+    @GET
+    public CompletionStage<List<ReactiveBookEntity>> getBooks(@QueryParam("sort") String sort) {
+        if (sort != null) {
+            return ReactiveBookEntity.listAll(Sort.ascending(sort));
+        }
+        return ReactiveBookEntity.listAll();
+    }
+
+    @GET
+    @Path("/stream")
+    @Produces(MediaType.SERVER_SENT_EVENTS)
+    @SseElementType(MediaType.APPLICATION_JSON)
+    public Publisher<ReactiveBookEntity> streamBooks(@QueryParam("sort") String sort) {
+        if (sort != null) {
+            return ReactiveBookEntity.streamAll(Sort.ascending(sort));
+        }
+        return ReactiveBookEntity.streamAll();
+    }
+
+    @POST
+    public CompletionStage<Response> addBook(ReactiveBookEntity book) {
+        return book.persist().thenApply(v -> {
+            //the ID is populated before sending it to the database
+            String id = book.id.toString();
+            return Response.created(URI.create("/books/entity" + id)).build();
+        });
+    }
+
+    @PUT
+    public CompletionStage<Response> updateBook(ReactiveBookEntity book) {
+        return book.update().thenApply(v -> Response.accepted().build());
+    }
+
+    // PATCH is not correct here but it allows to test persistOrUpdate without a specific subpath
+    @PATCH
+    public CompletionStage<Response> upsertBook(ReactiveBookEntity book) {
+        return book.persistOrUpdate().thenApply(v -> Response.accepted().build());
+    }
+
+    @DELETE
+    @Path("/{id}")
+    public CompletionStage<Void> deleteBook(@PathParam("id") String id) {
+        return ReactiveBookEntity.findById(new ObjectId(id)).thenCompose(book -> book.delete());
```

Comment:
We should really add a `deleteById(id)` operation, no?

Reply:
I would love to, and in both MongoDB and Hibernate Panache!
I'll open an issue for it so we will see if someone else would love it :)

Reply:
See the PR here: #6270